pillar-users.sls

```diff
@@ -0,0 +1,23 @@
+{# We expect 'docker-users' to yield a dictionary, where each key maps to a list of usernames.
+   The dictionary is the result of multiple merged pillar files targeted to this minion. The
+   value() operation turns the dict into a list of lists of users. The sum(start=[]) filter
+   is a fold that flattens this into a plain list of users and the unique filter removes
+   potentially occurring duplicates.
+
+   It is the users obligation to ensure that all users in `dockerusers` are actually present
+   on this minion. Since we do not know the way in which users are placed on minions, i.e.
+   thorugh which states, we cannot enforce this here with include directives.
+#}
```

Comment:
General note: I think it looks nicer to have YAML comments instead of Jinja ones.

```yaml
# some multiline
# comment in YAML
# so much nicer than

{#
  Jinja's way
  of doing things
#}
```

As the comments below are also YAML comments, I suggest to use them consistently. (Also, at least my editor can cope with the YAML comments better, e.g. wenn automatically commenting/ uncommenting multiple lines). Any particular reason you chose Jinja here?
(obviously, this is oppinionated style, so no hard veto on this)